src/noop.jl

```diff
@@ -53,16 +53,18 @@ Note that this method may return a wrong position when
 - some data have been inserted by `TranscodingStreams.unread`, or
 - the position of the wrapped stream has been changed outside of this package.
 """
-function Base.position(stream::NoopStream)
+function Base.position(stream::NoopStream)::Int64
     mode = stream.state.mode
-    if mode === :idle
-        return Int64(0)
+    if !isopen(stream)
+        throw_invalid_mode(mode)
+    elseif mode === :idle
+        return 0
+    elseif has_sharedbuf(stream)
+        return position(stream.stream)
```

Comment:
Here I treat the buffer as being "owned" by the underlying stream even though it is being shared. This is also why I need to add special cases for writing to a `NoopStream` that shares its buffer with the underlying stream. The underlying stream must be notified to change from `:idle` to `:write` mode before its `buffer1` is changed.